modules/gdscript/gdscript.cpp

```diff
@@ -1372,6 +1372,35 @@ String GDScript::debug_get_script_name(const Ref<Script> &p_script) {
 }
 #endif
 
+thread_local GDScript::UpdatableFuncPtr GDScript::func_ptrs_to_update_thread_local;
+
+GDScript::UpdatableFuncPtrElement GDScript::_add_func_ptr_to_update(GDScriptFunction **p_func_ptr_ptr) {
+	func_ptrs_to_update_thread_local.mutex.lock();
+	UpdatableFuncPtrElement result{ func_ptrs_to_update_thread_local.ptrs.push_back(p_func_ptr_ptr), &func_ptrs_to_update_thread_local.mutex };
+
+	if (unlikely(!func_ptrs_to_update_thread_local.initialized)) {
+		func_ptrs_to_update_thread_local.initialized = true;
+		// Unlock thread local mutex before main one to prevent a deadlock
+		func_ptrs_to_update_thread_local.mutex.unlock();
+
+		func_ptrs_to_update_mutex.lock();
+		func_ptrs_to_update.push_back(&func_ptrs_to_update_thread_local);
+		func_ptrs_to_update_mutex.unlock();
```

Comment:
```suggestion
		MutexLock lock(func_ptrs_to_update_mutex);
		func_ptrs_to_update.push_back(&func_ptrs_to_update_thread_local);
```
I'd suggest using `MutexLock` here and in places where the locking is scope based, it ensures some degrees of safety and avoids forgetting to unlock it.